src/test/jtx/TrustedPublisherServer.h

```diff
@@ -118,26 +229,212 @@ class TrustedPublisherServer
+        pushd /tmp/__certs__
+        rm *.crt *.key *.pem
+
+        # generate CA
+        openssl genrsa -out ca.key 2048
+        openssl req -new -x509 -nodes -days 10000 -key ca.key -out ca.crt \
+            -subj "/C=US/ST=CA/L=Los Angeles/O=rippled-unit-tests/CN=example.com"
+        # generate private cert
+        openssl genrsa -out server.key 2048
+        # Generate certificate signing request
+        # since our unit tests can run in either ipv4 or ipv6 mode,
+        # we need to use extensions (subjectAltName) so that we can
+        # associate both ipv4 and ipv6 localhost addresses with this cert
+        cat >"extras.cnf" <<EOF
+        [req]
+        req_extensions = v3_req
+        distinguished_name = req_distinguished_name
+
+        [req_distinguished_name]
+
+        [v3_req]
+        subjectAltName = @alt_names
+
+        [alt_names]
+        DNS.1 = localhost
+        IP.1 = ::1
+        EOF
+        openssl req -new -key server.key -out server.csr \
+            -config extras.cnf \
+            -subj "/C=US/ST=California/L=San Francisco/O=rippled-unit-tests/CN=127.0.0.1" \
+
+        # Create public certificate by signing with our CA
+        openssl x509 -req -days 10000 -in server.csr -CA ca.crt -CAkey ca.key -out server.crt \
+            -extfile extras.cnf -set_serial 01 -extensions v3_req
+
+        # generate DH params for server
+        openssl dhparam -out dh.pem 2048
+        # verify certs
+        openssl verify -CAfile ca.crt server.crt
+        openssl x509 -in server.crt -text -noout
+        popd
+     ```
+    */
+    static
+    std::string const&
+    cert()
+    {
+        static std::string const cert {R"cert(
+-----BEGIN CERTIFICATE-----
+MIIDczCCAlugAwIBAgIBATANBgkqhkiG9w0BAQUFADBjMQswCQYDVQQGEwJVUzEL
+MAkGA1UECAwCQ0ExFDASBgNVBAcMC0xvcyBBbmdlbGVzMRswGQYDVQQKDBJyaXBw
+bGVkLXVuaXQtdGVzdHMxFDASBgNVBAMMC2V4YW1wbGUuY29tMB4XDTE5MDgwNzE3
+MzM1OFoXDTQ2MTIyMzE3MzM1OFowazELMAkGA1UEBhMCVVMxEzARBgNVBAgMCkNh
+bGlmb3JuaWExFjAUBgNVBAcMDVNhbiBGcmFuY2lzY28xGzAZBgNVBAoMEnJpcHBs
+ZWQtdW5pdC10ZXN0czESMBAGA1UEAwwJMTI3LjAuMC4xMIIBIjANBgkqhkiG9w0B
+AQEFAAOCAQ8AMIIBCgKCAQEA5Ky0UE9K+gFOznfwBvq2HfnQOOPGtVf4G9m63b5V
+QNJYCSNiYxkGZW72ESM3XA8BledlkV9pwIm17+7ucB1Ed3efQjQDq2RSk5LDYDaa
+r0Qzzy0EC3b9+AKA6mAoVY6s1Qws/YvM4esz0H+SVvtVcFqA46kRWhJN7M5ic1lu
+d58fAq04BHqi5zOEOzfHJYPGUgQOxRTHluYkkkBrL2xioHHnOROshW+PIYFiAc/h
+WPzuihPHnKaziPRw+O6O8ysnCxycQHgqtvx73T52eJdLxtr3ToRWaY/8VF/Cog5c
+uvWEtg6EucGOszIH8O7eJWaJpVpAfZIX+c62MQWLpOLi/QIDAQABoyowKDAmBgNV
+HREEHzAdgglsb2NhbGhvc3SHEAAAAAAAAAAAAAAAAAAAAAEwDQYJKoZIhvcNAQEF
+BQADggEBAOhLAO/e0lGi9TZ2HiVi4sJ7KVQaBQHGhfsysILoQNHrNqDypPc/ZrSa
+WQ2OqyUeltMnUdN5S1h3MKRZlbAeBQlwkPdjTzlzWkCMWB5BsfIGy5ovqmNQ7zPa
+Khg5oxq3mU8ZLiJP4HngyU+hOOCt5tttex2S8ubjFT+3C3cydLKEOXCUPspaVkKn
+Eq8WSBoYTvyUVmSi6+m6HGiowWsM5Qgj93IRW6JCbkgfPeKXC/5ykAPQcFHwNaKT
+rpWokcavZyMbVjRsbzCQcc7n2j7tbLOu2svSLy6oXwG6n/bEagl5WpN2/TzQuwe7
+f5ktutc4DDJSV7fuYYCuGumrHAjcELE=
+-----END CERTIFICATE-----
+)cert"};
```

Comment:
nit: this might look better indented.

Reply:
easier to paste from command line if we don't indent and leave it as a raw string block. I don't think leading spaces would be permitted by the current raw string decl (or openssl wouldn't like it..). That said, another  approach would be to figure out what openssl api calls could generate these strings at runtime (the equiv of the command lines I noted in the comment) and then we don't need the pasted wall of text at all.